x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_view/category_stopped_partitions.tsx

```diff
@@ -0,0 +1,159 @@
+import { ml } from '../../../../../../../services/ml_api_service';
+import { extractErrorProperties } from '../../../../../../../../../common/util/errors';
+
+const NUMBER_OF_PREVIEW = 5;
+export const CategoryStoppedPartitions: FC = () => {
+  const { jobCreator: jc, resultsLoader } = useContext(JobCreatorContext);
+  const jobCreator = jc as CategorizationJobCreator;
+  const [tableRow, setTableRow] = useState<Array<{ partitionName: string }>>([]);
+  const [hasStoppedPartitions, setHasStoppedPartitions] = useState(false);
+  const [stoppedPartitionsError, setStoppedPartitionsError] = useState<string | undefined>();
+
+  // using ref so this Subscription instance is only intialized once
+  const _resultsSubscription: MutableRefObject<Subscription | undefined> = useRef();
+
+  const columns = useMemo(
+    () => [
+      {
+        field: 'partitionName',
+        name: i18n.translate(
+          'xpack.ml.newJob.wizard.pickFieldsStep.stoppedPartitionsPreviewColumnName',
+          {
+            defaultMessage: 'Stopped partition names',
+          }
+        ),
+        render: (partition: any) => (
+          <EuiText size="s">
+            <code>{partition}</code>
+          </EuiText>
+        ),
+      },
+    ],
+    []
+  );
+  function setResultsWrapper(results: Results) {
+    loadCategoryStoppedPartitions();
+  }
+
+  async function loadCategoryStoppedPartitions() {
+    try {
+      const results = await ml.results.getCategoryStoppedPartitions([jobCreator.jobId]);
+
+      if (
+        results?.jobs !== undefined &&
+        !Array.isArray(results?.jobs) && // if jobs is object of jobId: [partitions]
+        Array.isArray(results?.jobs[jobCreator.jobId]) &&
+        results.jobs[jobCreator.jobId].length > 0
+      ) {
+        const stoppedPartitionsPreview = results.jobs[jobCreator.jobId];
+        // once we have reached number of stopped partitions we wanted to show as preview
+        // no need to keep fetching anymore
+        if (
+          stoppedPartitionsPreview.length >= NUMBER_OF_PREVIEW &&
+          _resultsSubscription &&
+          _resultsSubscription.current !== undefined
+        ) {
+          _resultsSubscription.current.unsubscribe();
+          _resultsSubscription.current = undefined;
+        }
+        setHasStoppedPartitions(true);
+        setTableRow(
+          stoppedPartitionsPreview.slice(0, NUMBER_OF_PREVIEW).map((partitionName) => ({
+            partitionName,
+          }))
+        );
+      }
+    } catch (e) {
+      const error = extractErrorProperties(e);
+      // might get 404 because job has not been created yet and that's ok
+      if (error.statusCode !== 404) {
+        setStoppedPartitionsError(error.message);
+      }
+    }
+  }
+
+  useEffect(() => {
+    // only need to run this check if jobCreator.perPartitionStopOnWarn is turned on
+    if (
+      jobCreator.perPartitionCategorization &&
+      jobCreator.perPartitionStopOnWarn &&
+      _resultsSubscription?.current
```

Comment:
this check stops the subscription from being created because `_resultsSubscription?.current` is `null` initially

Reply:
Thanks. I removed this whole block and replaced it with the rxjs utils instead here https://github.com/elastic/kibana/pull/75709/commits/63a4e02cb936ac5f32538adc6d0e2ffcf527c034